frontend/src/scenes/Auth/components/ResetPassword/index.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { FC, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { Formik } from 'formik';
+import { Button, Form, InputGroup } from 'react-bootstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faEye, faEyeSlash } from '@fortawesome/free-solid-svg-icons';
+import { IBindingCallback1 } from '../../../../common/models/callback/IBindingCallback1';
+import { IResetPasswordCallback } from '../../../../common/models/auth/IResetPasswordCallback';
+import { Routes } from '../../../../common/enums/Routes';
+import resetPasswordSchema from '../../../../common/models/formik/resetPasswordSchema';
+import { THandleResetPassword } from '../../../../common/models/auth/THandleResetPassword';
+import initialValuesResetPassword from '../../../../common/models/formik/initialValuesResetPassword';
+
+import styles from './styles.module.scss';
+
+interface IProps {
+  resetPassword: IBindingCallback1<IResetPasswordCallback>;
+  token: string
+}
+
+const ResetPassword: FC<IProps> = ({ resetPassword, token }) => {
+  const history = useHistory<string>();
+  const [isPassword, setPassword] = useState(true);
+  const [isConfirmPassword, setConfirmPassword] = useState(true);
+  const handleResetPassword: THandleResetPassword = async ({ password }) => {
+    await resetPassword({ token, password });
+    history.push(Routes.SignIn);
+  };
```

Comment:
no async functions in components, all async operations should be handled via sagas

Reply:
Solved